src/main/webapp/assets/js/feedScript.js

```diff
@@ -627,17 +622,34 @@ async function submitModal() {
 
     let url;
 
-    if (latLngResult) {
-      const lat = latLngResult.lat;
-      const lng = latLngResult.long;
-      url = '/postData?' + 'collegeId=' + collegeId + '&lat=' + lat + '&lng=' + lng;
+    // If the address entered isn't the same one as the one we last checked,
+    // then we can assume the user edited their address and we should display
+    // a fresh error if this new address is also invalid.
+    if (!latLngResult && modalLocation !== cachedModalAddress) {
+      cachedModalAddress = modalLocation;
+      const modalError = document.createElement('div');
+      modalError.setAttribute('id', 'modal-error');
+      modalError.innerText =
+        `We couldn't find address '${modalLocation}'. ` +
+        'Please check your address for errors. ' +
+        'If you wish to submit anyway, no pin will be added to the map.';
+      document.getElementById('modal-form')
+          .insertBefore(modalError, document.getElementById('modal-submit'));
+    // Else, if the invalid address is the same as we last checked
+    // or the address is just plain valid, then add the post to the Datastore.
     } else {
-      url = '/postData?' + 'collegeId=' + collegeId + '&lat=0&lng=0';
+      const modalError = document.getElementById('modal-error');
+      if (modalError) {
+        document.getElementById('modal-form').removeChild(modalError);
+      }
+
+      const lat = latLngResult ? latLngResult.lat : 0;
+      const lng = latLngResult ? latLngResult.long : 0;
```

Comment:
Would this still put the marker at (0,0) ?

Reply:
Yes, I was thinking about what values to transmit to indicate an 'invalid' or 'nonexistant' lat/long, and here's what I came up with:

**Use (0,0) and don't show (0,0) values on the map**
Pros:
- Doesn't break number parsing, since it is a number
- Simple to implement
- Is the only '[falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy)' number, so it's easy to check for valid lat/longs (see below:)
https://github.com/googleinterns/noms/blob/fd2adb3e02fc561a481c3a60c25bde59288d7bba/src/main/webapp/assets/js/feedScript.js#L407
- There's no valid reason why someone would want to actually locate an event at (0,0), given that non-U.S. locations are out of scope for our project and the fact that (0, 0) is in the Atlantic ocean.

Cons:
- Possibly worse for code readability? I will add a comment that clarifies (0, 0) = invalid.

**Use `null` or `undefined` and don't show posts with those lat/longs on the map**
Pros:
- Code is crystal clear

Cons:
- Breaks number parsing, so the servlet and entire chain of transmission needs to be edited to allow this value
- Harder to implement

**Use a boolean to indicate whether a post has a valid lat/long**
Pros:
- ?

Cons:
- More checks will have to be implemented since some posts will have lat/longs
- Is really just the previous suggestion with more variables that might become out of sync
- Will still have to put _something_ in the lat/long field or else it will default to `undefined`
- Yeah this one wasn't a good idea

Reply:
Oh I understand now! I forgot that 0 also means false. I guess one thing to also keep in mind is that if the lat or lng is 0, it would read as no location, but neither of those are in the US so it is outside this scope.